### ISSUE TITLE
[Feature]: Allow reordering metrics by which runs can be sorted

### ISSUE THREAD

**Issue by tristanengst** [wandb/wandb#6690]
### Description

Presently, the metrics by which runs can be sorted can not be reordered. Suppose first sort by hyperparameter `A`, then hyperparameter `B`, and finally hyperparameter `C`. I now introduce another hyperparameter `D`, but I wish to sort in the following order: `A, D, B, C`. To accomplish this, I would need to change the second, and third sorting metrics from `B` to `D` and `C` to `B` respectively, and then add a new sort by hyperparameter `C` at the end (three total actions). The number of actions is the number of sorting metrics that need to be changed (how far up the list I want to insert the new sorting metric) plus one. It's thus quite annoying to add new metrics to sort by.

In this image, you can see such an example—imagine if I want to add a sort by weight decay towards the top of the list of sorts.
![image](https://github.com/wandb/wandb/assets/3527106/cf21aed6-dcff-4b03-bf12-2b378cd8c62a)

### Suggested Solution

Have a click-and-drag button to each of the sorting metrics, so they can be created by clicking `Add another field` and the dragging the resulting sorting metric to where it is desired . Thus the cost of inserting a new one is at most two actions. The button could go to the left of each sort, while moving the `x` that deletes the sort to the right of the sort to mitigate the number of accidental clicks, and to be consistent with the Filter drop down.
